testsuite/cluster-test/src/experiments/versioning_test.rs

```diff
@@ -0,0 +1,264 @@
+    full_nodes: Vec<Instance>,
+    updated_image_config: InstanceConfig,
+    base_image_config: InstanceConfig,
+    full_node_config: InstanceConfig,
+}
+
+impl ExperimentParam for ValidatorVersioningParams {
+    type E = ValidatorVersioning;
+    fn build(self, cluster: &Cluster) -> Self::E {
+        if self.count > cluster.validator_instances().len() {
+            panic!(
+                "Can not reboot {} validators in cluster with {} instances",
+                self.count,
+                cluster.validator_instances().len()
+            );
+        }
+        let mut instances = Vec::with_capacity(self.count);
+        let mut all_instances = cluster.validator_instances().to_vec();
+        let mut rnd = rand::thread_rng();
+        for _i in 0..self.count {
+            let instance = all_instances.remove(rnd.gen_range(0, all_instances.len()));
+            instances.push(instance);
+        }
+        let base_image_config = cluster
+            .random_validator_instance()
+            .instance_config()
+            .clone();
+        let mut updated_image_config = base_image_config.clone();
+        updated_image_config
+            .replace_tag(self.updated_image_tag.clone())
+            .unwrap();
+
+        let mut full_node_config = cluster
+            .fullnode_instances()
+            .get(0)
+            .unwrap()
+            .instance_config()
+            .clone();
+        full_node_config
+            .replace_tag(self.updated_image_tag)
+            .unwrap();
+
+        Self::E {
+            first_batch: instances,
+            second_batch: all_instances,
+            full_nodes: cluster.fullnode_instances().to_vec(),
+            full_node_config,
+            updated_image_config,
+            base_image_config,
+        }
+    }
+}
+
+#[async_trait]
+impl Experiment for ValidatorVersioning {
+    fn affected_validators(&self) -> HashSet<String> {
+        instance::instancelist_to_set(&self.first_batch)
+            .union(&instance::instancelist_to_set(&self.second_batch))
+            .cloned()
+            .collect()
+    }
+
+    async fn run(&mut self, context: &mut Context<'_>) -> anyhow::Result<()> {
+        // Mint a number of accounts
+        context
+            .tx_emitter
+            .mint_accounts(
+                &EmitJobRequest::for_instances(
+                    context.cluster.validator_instances().to_vec(),
+                    context.global_emit_job_request,
+                ),
+                15,
+            )
+            .await?;
+
+        // Change the images for the instances in the first batch
+        let futures: Vec<_> = self.first_batch.iter().map(Instance::stop).collect();
+        try_join_all(futures).await?;
+        let futures: Vec<_> = self
+            .first_batch
+            .iter()
+            .map(|_| {
+                context
+                    .cluster_swarm
+                    .spawn_new_instance(self.updated_image_config.clone(), false)
```

Comment:
Note InstanceConfig contain things like validator index, etc.

So you would ned to update config for each instance individually, otherwise you are just starting copy of the same instance multiple times